core/constraints/validation.go

```diff
@@ -215,20 +219,79 @@ func (v *validator) checkValidValues(cons Value) error {
 	return nil
 }
 
+// InvalidVocabValueErr represents an error that occurs
+// when a validation constraint is violated.
+// It provides details for the
+// invalid inputs, closest valid values and all possible valid values.
+type InvalidVocabValueErr struct {
```

Comment:
Prefer full names for types.

```suggestion
type InvalidVocabValueError struct {
```

Reply:
Updated, and noted with thanks Harry!